nomad/grapl-core.nomad

```diff
@@ -415,179 +433,217 @@ job "grapl-core" {
 
   group "grapl-node-identifier" {
     count = var.num_node_identifiers
 
-    network {
-      mode = "bridge"
+    //    network {
+    //      mode = "bridge"
+    //    }
+
+    task "provisioner" {
+      driver = "docker"
+
+      config {
+        image        = "${var.container_registry}/grapl/provisioner:${var.provisioner_tag}"
+        network_mode = "grapl-network"
+      }
+
+      lifecycle {
+        hook    = "prestart"
+        sidecar = false
+      }
+
+      env {
+        GRAPL_AWS_ENDPOINT            = var.aws_endpoint
+        GRAPL_AWS_ACCESS_KEY_ID       = var.aws_access_key_id
+        GRAPL_AWS_ACCESS_KEY_SECRET   = var.aws_access_key_secret
+        AWS_DEFAULT_REGION            = var.aws_region # boto3 prefers this one
+        AWS_REGION                    = var.aws_region
+        GRAPL_SCHEMA_TABLE            = var.schema_table_name
+        GRAPL_SCHEMA_PROPERTIES_TABLE = var.schema_properties_table_name
+        GRAPL_USER_AUTH_TABLE         = var.user_auth_table
+        GRAPL_TEST_USER_NAME          = var.grapl_test_user_name
+      }
+
     }
 
     task "node-identifier" {
       driver = "docker"
 
       config {
-        image = "${var.container_registry}/grapl/node-identifier:${var.node_identifier_tag}"
+        image        = "${var.container_registry}/grapl/node-identifier:${var.node_identifier_tag}"
+        network_mode = "grapl-network"
       }
 
       env {
-        RUST_LOG                    = "${var.rust_log}"
-        REDIS_ENDPOINT              = "${var.redis_endpoint}"
-        MG_ALPHAS                   = "${local.alpha_grpc_connect_str}"
-        GRAPL_SCHEMA_TABLE          = "${var.schema_table_name}"
-        AWS_REGION                  = "${var.aws_region}"
-        GRAPL_DYNAMIC_SESSION_TABLE = "${var.session_table}"
+        GRAPL_AWS_ENDPOINT          = var.aws_endpoint
+        GRAPL_AWS_ACCESS_KEY_ID     = var.aws_access_key_id
+        GRAPL_AWS_ACCESS_KEY_SECRET = var.aws_access_key_secret
+        AWS_DEFAULT_REGION          = var.aws_region # boto3 prefers this one
+        AWS_REGION                  = var.aws_region
+        RUST_LOG                    = var.rust_log
+        RUST_BACKTRACE              = 1
+        REDIS_ENDPOINT              = var.redis_endpoint
+        MG_ALPHAS                   = local.alpha_grpc_connect_str
+        GRAPL_SCHEMA_TABLE          = var.schema_table_name
+        GRAPL_DYNAMIC_SESSION_TABLE = var.session_table_name
         # https://github.com/grapl-security/grapl/blob/18b229e824fae99fa2d600750dd3b17387611ef4/pulumi/grapl/__main__.py#L156
-        DEST_BUCKET_NAME      = "${var.subgraphs_generated_bucket}"
-        SOURCE_QUEUE_URL      = "${var.node_identifier_queue}"
-        DEAD_LETTER_QUEUE_URL = "${var.node_identifier_dead_letter_queue}"
+        DEST_BUCKET_NAME      = var.subgraphs_generated_bucket
+        SOURCE_QUEUE_URL      = var.node_identifier_queue
+        DEAD_LETTER_QUEUE_URL = var.node_identifier_dead_letter_queue
```

Comment:
This is also incorrect; the value should be the retry queue.

Reply:
Great call